### PR TITLE
Clean up docker containers/volumes after command execution

### PR DESCRIPTION
**Pull request by mitalipaygude**
*Issue #, if available:*
https://github.com/aws/eks-anywhere/issues/169

*Description of changes:*
Running the command multiple times on the same machine might result in volumes being leftover and fill up disk space.
Fixing that issue in this pr.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/332#" title="Author self-approved">mitalipaygude</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mitalipaygude]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/executables/executables.go
Comment (TerryHowe) [inline pkg/executables/executables.go:96]: This is probably a fine strategy, but seems like having process id in there would be a good too.

Comment (mitalipaygude) [reply pkg/executables/executables.go:96]: do you mean the container id for those docker processes?

Comment (TerryHowe) [reply pkg/executables/executables.go:96]: No, I meant the pid of the cli. Don't worry about it though, this is probably a good enough strategy.

File: pkg/executables/executables.go
Comment (vivek-koppuru) [inline pkg/executables/executables.go:182]: This should be a member method of `linuxDockerExecutable`

File: pkg/executables/executables.go
Comment (vivek-koppuru) [inline pkg/executables/executables.go:191]: Let's log the docker container name that we failed to clean up

File: pkg/executables/executables.go
Comment (vivek-koppuru) [inline pkg/executables/executables.go:187]: Let's include name here, not sure if the format is as below
```suggestion
	logger.V(6).Info("cleaning up command", "cmd", cmd, "name", containerName)
```

File: pkg/executables/executables.go
Comment (vivek-koppuru) [inline pkg/executables/executables.go:115]: Do we want to error if cleaning up fails? If we do, we can probably just not use defer and call it below with error blocks. Wondering if we want to surface the error in cleaning up with at a lesser log level, something like 1 or even 0.

File: pkg/executables/executables.go
Comment (vivek-koppuru) [inline pkg/executables/executables.go:96]: What if we use epoch timestamp? I am not sure how readable this is, but epoch would require a converter to actually get the time. This is also worth a read in the case that two cli's were running at the same time and accidentally picks the same time: https://stackoverflow.com/a/54491783

File: pkg/executables/executables.go
Comment (mitalipaygude) [reply pkg/executables/executables.go:187]: the command already shows the container name; for eg

`Executing clean up command	{"cmd": "/usr/local/bin/docker rm -f -v eksa_kubectl_2021-09-2813.25.27.128915"}`

File: pkg/executables/executables.go
Comment (mitalipaygude) [reply pkg/executables/executables.go:115]: I don't believe the cluster creation/deletion/updates should stop because we fail to clean up some docker containers or volumes. We could print a warning maybe that in case the container clean up fails multiple times, volumes can keep taking up more space and suggest the users to run something like `docker system prune -a`. We can bump the logging level for the clean ups to 1 or 0 if we want.

File: pkg/executables/executables.go
Comment (vivek-koppuru) [reply pkg/executables/executables.go:187]: Let's change the text tho

File: pkg/executables/executables.go
Comment (vivek-koppuru) [inline pkg/executables/executables.go:192]: If the verbosity is 0, do we need to specify this here? Also can we display what container failed to clean up here? Because of that, wondering if we should do verbosity of 1 instead.

**Review by vivek-koppuru — APPROVED** [review]
/lgtm

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/332#" title="Author self-approved">mitalipaygude</a>*, *<a href="https://github.com/aws/eks-anywhere/pull/332#pullrequestreview-766036147" title="LGTM">vivek-koppuru</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mitalipaygude,vivek-koppuru]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Pull request merged**
